metadata/metadata.go

```diff
@@ -832,6 +866,45 @@ func (resource *modelResource) UpdateSchedule(schedule string) error {
 	return fmt.Errorf("not implemented")
 }
 
+func (resource *modelResource) Update(lookup ResourceLookup) error {
+	val, err := lookup.Lookup(resource.ID())
+	if err != nil {
+		return err
+	}
+	deserialized := val.Proto()
+	model, ok := deserialized.(*pb.Model)
+	if !ok {
+		return errors.New("failed to deserialize existing model record")
+	}
+	updated, err := unionNameVariants(resource.serialized.Features, model.Features)
+	if err != nil {
+		return err
+	}
+	resource.serialized.Features = updated
+	updated, err = unionNameVariants(resource.serialized.Trainingsets, model.Trainingsets)
+	if err != nil {
+		return err
+	}
+	resource.serialized.Trainingsets = updated
+	return nil
+}
+
+func unionNameVariants(source, destination []*pb.NameVariant) ([]*pb.NameVariant, error) {
+	set := make(map[string]bool)
+
+	for _, variant := range destination {
```

Comment:
this is only using the variants not the names, you could make or us a struct NameVariant{name string, variant string} and make set := make(map[NameVariant]bool)

Reply:
I originally had the make use a `NameVariant` has a key, but it seems that the comparison for the map isn't performed by value, so I was never seeing updates.

Reply:
I see, would rename variant to nameVariant in this loop and the next loop, I got confused